code/game/objects/radiation.dm

```diff
@@ -31,8 +31,11 @@
 
 /mob/living/rad_act(amount, silent = 0)
 	if(amount)
-		var/message = silent ? null : "Your clothes feel warm."
-		var/blocked = run_armor_check(null, "rad", message, message)
+		var/blocked = getarmor(null, "rad")
+
+		if(!silent)
+			src << "Your skin feels warm."
```

Comment:
You're welcome to pull that line out and implement a print_rad_warning() proc for every type out there.